api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -208,15 +210,29 @@ public boolean removeMembership(CohortMembership cohortMembership) {
 	public Collection<CohortMembership> getActiveMemberships() {
 		return getActiveMemberships(new Date());
 	}
-	
+
 	/**
+	 * Searches for the {@code patients} membership and returns it.
+	 * <p>
+	 * If the patient has no membership, an empty Optional is returned.
+	 *
+	 * @param patient Holder of the searched membership.
+	 * @return Optional membership of the given patient.
+	 * @throws NoSuchElementException Thrown if the patient has no active membership.
 	 * @since 2.1.0
 	 */
 	public CohortMembership getActiveMembership(Patient patient) {
-		return getMemberships().stream().filter(m -> m.isActive() && m.getPatientId().equals(patient.getPatientId()))
```

Comment:
Actually the above line you removed looks fine. You are just expected to change the way data is returned/collected after filtration. That instead of first collecting all members that qualify by `.collect(Collectors.toList())` and then get the first one, just do something like `findFirst()` the get.

Reply:
Its about formatting and readability which is actually rather poor in this line.